File: prime.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

// Función para verificar si un número es primo
bool esPrimo(int numero) {
    if (numero < 2) return false;
    for (int i = 2; i * i <= numero; i++) {
        if (numero % i == 0) return false;
    }
    return true;
}

// Función para lanzar un dado (número aleatorio entre 1 y 6)
int lanzarDado() {
    return rand() % 6 + 1;
}

// Función para jugar una ronda
int jugarRonda() {
    int dado1 = lanzarDado();
    int dado2 = lanzarDado();
    int total = dado1 + dado2;

    cout << "Primer lanzamiento: " << dado1 << " y " << dado2 << " (Total: " << total << ")" << endl;

    // Reglas especiales
    if (dado1 == dado2) {
        // Dobles: lanzar 3 dados adicionales
        cout << "Dobles! Lanzando 3 dados adicionales..." << endl;
        for (int i = 0; i < 3; i++) {
            total += lanzarDado();
        }
        cout << "Nuevo total después de lanzar 3 dados adicionales: " << total << endl;
    } else if (total == 12) {
        // Preguntar si desea lanzar un dado adicional cuando el total es 12
        char opcion;
        cout << "Has sacado un total de 12! ¿Deseas lanzar un dado adicional? (s/n): ";
        cin >> opcion;
        if (opcion == 's' || opcion == 'S') {
            total += lanzarDado();
            cout << "Nuevo total después de lanzar 1 dado adicional: " << total << endl;
        }
    }

    return total;
}

// Función principal del juego
void jugar() {
    srand(time(0));
    char jugarDeNuevo = 's';

    while (jugarDeNuevo == 's' || jugarDeNuevo == 'S') {
        int totalJugador1 = 0, totalJugador2 = 0;
        bool empate = true;

        while (empate) {
            cout << "Jugador 1 juega:" << endl;
            totalJugador1 = jugarRonda();
            cout << "Jugador 2 juega:" << endl;
            totalJugador2 = jugarRonda();

            cout << "Jugador 1 total: " << totalJugador1 << " | Jugador 2 total: " << totalJugador2 << endl;

            // Verificar si hay empate
            if (totalJugador1 != totalJugador2) {
                empate = false;
            } else {
                cout << "Empate! Repitiendo la ronda..." << endl;
            }
        }

        // Determinar el ganador
        if (esPrimo(totalJugador1) && esPrimo(totalJugador2)) {
            if (totalJugador1 > totalJugador2) {
                cout << "Jugador 1 gana con un número primo!" << endl;
            } else {
                cout << "Jugador 2 gana con un número primo!" << endl;
            }
        } else if (esPrimo(totalJugador1)) {
            cout << "Jugador 1 gana con un número primo!" << endl;
        } else if (esPrimo(totalJugador2)) {
            cout << "Jugador 2 gana con un número primo!" << endl;
        } else if (totalJugador1 > totalJugador2) {
            cout << "Jugador 1 gana!" << endl;
        } else {
            cout << "Jugador 2 gana!" << endl;
        }

        // Preguntar si desean jugar de nuevo
        cout << "¿Quieres jugar de nuevo? (s/n): ";
        cin >> jugarDeNuevo;
    }
}

int main() {
    cout << "Bienvenido a Prime Jack o 23!" << endl;
    jugar();
    return 0;
}

File: prime.py
import random

# Función para verificar si un número es primo
def es_primo(numero):
    if numero < 2:
        return False
    for i in range(2, int(numero ** 0.5) + 1):
        if numero % i == 0:
            return False
    return True

# Función para lanzar un dado (número aleatorio entre 1 y 6)
def lanzar_dado():
    return random.randint(1, 6)

# Función para jugar una ronda
def jugar_ronda():
    dado1 = lanzar_dado()
    dado2 = lanzar_dado()
    total = dado1 + dado2

    print(f"Primer lanzamiento: {dado1} y {dado2} (Total: {total})")

    # Reglas especiales
    if dado1 == dado2:
        # Dobles: lanzar 3 dados adicionales
        print("Dobles! Lanzando 3 dados adicionales...")
        for _ in range(3):
            total += lanzar_dado()
        print(f"Nuevo total después de lanzar 3 dados adicionales: {total}")
    elif total == 12:
        # Preguntar si desea lanzar un dado adicional cuando el total es 12
        opcion = input("Has sacado un total de 12! ¿Deseas lanzar un dado adicional? (s/n): ").lower()
        if opcion == 's':
            total += lanzar_dado()
            print(f"Nuevo total después de lanzar 1 dado adicional: {total}")

    return total

# Función principal del juego
def jugar():
    jugar_de_nuevo = 's'

    while jugar_de_nuevo == 's':
        total_jugador1 = 0
        total_jugador2 = 0
        empate = True

        # Jugar hasta que no haya empate
        while empate:
            print("Jugador 1 juega:")
            total_jugador1 = jugar_ronda()
            print("Jugador 2 juega:")
            total_jugador2 = jugar_ronda()

            print(f"Jugador 1 total: {total_jugador1} | Jugador 2 total: {total_jugador2}")

            # Verificar si hay empate
            if total_jugador1 != total_jugador2:
                empate = False
            else:
                print("Empate! Repitiendo la ronda...")

        # Determinar el ganador
        if es_primo(total_jugador1) and es_primo(total_jugador2):
            if total_jugador1 > total_jugador2:
                print("Jugador 1 gana con un número primo!")
            else:
                print("Jugador 2 gana con un número primo!")
        elif es_primo(total_jugador1):
            print("Jugador 1 gana con un número primo!")
        elif es_primo(total_jugador2):
            print("Jugador 2 gana con un número primo!")
        elif total_jugador1 > total_jugador2:
            print("Jugador 1 gana!")
        else:
            print("Jugador 2 gana!")

        # Preguntar si desean jugar de nuevo
        jugar_de_nuevo = input("¿Quieres jugar de nuevo? (s/n): ").lower()

# Inicia el juego directamente
print("Bienvenido a Prime Jack o 23!")
jugar()
